include/podio/Frame.h

```diff
@@ -234,8 +234,9 @@ class Frame {
   ///              is supported by GenericParameters
   /// @param key   The name under which this parameter should be stored
   /// @param value The value of the parameter. A copy will be put into the Frame
-  template <typename T, typename = podio::EnableIfValidGenericDataType<T>>
+  template <typename T>
   inline void putParameter(const std::string& key, T value) {
+    static_assert(podio::isSupportedGenericDataType<T>, "The type is not supported by the GenericParameters");
```

Comment:
```suggestion
    static_assert(podio::isSupportedGenericDataType<T>, "T is not a supported parameter type");
```
I think `GenericParameters` is too much of an implementation detail here, especially if users get this message from their algorithms, they will not know what it is and where to look. I would change this message at least for the cases in `Frame`. Not sure if we should also put the supported types in, because this message is likely to become outdated if we have to add more types.

Reply:
For the same reason `T` doesn't tell you much where to look or what to do. But showing the type isn't completely trivial :shrug:

Reply:
We need to have documentation where we state the supported datatypes (not sure if we already have). I think we could also drop the `T` from the message. In the end the compiler error will contain the type they tried to use in some form.

Reply:
The docs mention int, float, strong and vectors of those https://github.com/AIDASoft/podio/blob/master/doc/frame.md#usage-for-parameters

Reply:
Ah right, I knew it was somewhere :D. But it's missing a `double` :see_no_evil: